src/services/code_service.js

```
@@ -1,21 +1,33 @@
 function until(arr, fn) {
-	const l = arr.length;
-	let i = 0;
-	while (i < l && !fn(arr[i])) {
-		i++;
-	}
-	return arr.slice(0, i);
+  const l = arr.length;
+  let i = 0;
+  while (i < l && !fn(arr[i])) {
+    i++;
+  }
+  return arr.slice(0, i);
 }
 
+function ensure(condition, message) {
+  if (typeof condition === 'function') {
```

else呢？

---

condition不是function直接返回了。你觉得呢？应该throw error？

---

直接返回意味着静悄悄地失败；如果认为这个函数是内部的，可以不必检测参数。

---

我觉得silent failure不如throw error更好。它本身是为了Enforce契约的，自己也应该用契约约束输入参数。改成，如果不是`function`就throw error了。